src/software/new_geom/line.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <stdexcept>
+
+#include "software/new_geom/polynomial.h"
+
+class Line final : public Polynomial
+{
+   public:
+    /**
+     * Creates a line with zero y-intercept and zero slope
+     */
+    inline explicit Line() {}
+
+    /**
+     * Creates a line from a y-intercept and a slope
+     *
+     * @param y_intercept the y_intercept of the line
+     * @param slope the slope of the line
+     */
+    inline explicit Line(double y_intercept, double slope)
```

Comment:
What was the reason for inlining the class? I'd generally prefer a separate `.cpp` for consistency

Reply:
I did it because it's a pretty short class, but I'll make a separate `.cpp` file.

Reply:
Addressed in 746b91a